### PR TITLE
Feature/folders applets api

### PR DESCRIPTION
**Pull request by nmialik**

### CODE REVIEW

File: src/redux/modules/Account/Account.thunk.ts [hunk 1]
```diff
@@ -26,3 +33,33 @@ export const updateAlertStatus = createAsyncThunk(
...
 );
+
+export const getAppletsForFolders = createAsyncThunk(
+  'account/getAppletsInFolders',
+  async ({ account }: { account: Account }, { dispatch }) => {
+    const foldersData = await Promise.allSettled(
+      account.folders.map(async (folder) => await dispatch(getAppletsForFolder({ folder }))),
+    );
+
+    const result = [
```
Comment (IhorNesterenko99) [inline src/redux/modules/Account/Account.thunk.ts:44]: Could you please make return here?

File: src/redux/modules/Account/Account.utils.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import { Folder } from 'redux/modules';
+
+export type AppletResponse = {
```
Comment (IhorNesterenko99) [inline src/redux/modules/Account/Account.utils.ts:3]: Pls move types to the Account.schema.ts or to the Account.types.ts

File: src/redux/modules/Account/Account.utils.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        isExpanded: false,
+        isFolder: false,
+        isChild: true,
+      };
+    });
+
+    loadedFolder.items = items;
+    removeDuplicateApplets(loadedFolder);
+  }
+  return loadedFolder;
```
Comment (IhorNesterenko99) [inline src/redux/modules/Account/Account.utils.ts:99]: Pls add empty line before return statement

File: src/redux/modules/Account/Account.reducer.ts [hunk 4]
```diff
@@ -31,4 +31,34 @@ export const extraReducers = (builder: ActionReducerMapBuilder<AccountSchema>):
       switchAccount.error = error.response?.data as AxiosError<ErrorResponse>;
     }
   });
+
+  builder.addCase(getAppletsForFolders.pending, ({ accountFoldersApplets }, action) => {
+    if (accountFoldersApplets.status !== 'loading') {
```
Comment (kdanilenka32) [inline src/redux/modules/Account/Account.reducer.ts:36]: Would be great to move  these functions, we have similar functions in 2 places

File: src/redux/modules/Account/Account.thunk.ts (hunk 1 above)
Comment (IhorNesterenko99) [reply src/redux/modules/Account/Account.thunk.ts]: I mean without result variable: return [...

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**